SPECS/zstd/zstd.spec

```diff
@@ -1,96 +1,323 @@
+
+# enable .gz support by default
+%bcond_without zlib
+
+# enable pzstd support by default
+%bcond_without pzstd
+
+# Our tooling/pipelines still use the old rpm version 4.17 which doesn't understand this,
+# and I can't figure out how to fix the tooling/pipelines, so just avoid using bcond for now
+%bcond_with gtest
+
 Name:           zstd
 Version:        1.5.5
-Release:        2%{?dist}
+Release:        5%{?dist}
+Summary:        Zstd compression library
+
 License:        BSD-3-Clause AND GPL-2.0-only
 Vendor:         Microsoft Corporation
-Distribution:   Mariner
-Group:          Applications/File
-URL:            https://facebook.github.io/zstd/
+Distribution:   Microsoft Azure Linux
+URL:            https://github.com/facebook/zstd
 Source0:        https://github.com/facebook/zstd/releases/download/v%{version}/%{name}-%{version}.tar.gz
-Requires:       %{name}-libs = %{version}-%{release}
+
+Patch1:         pzstd.1.patch
+
+BuildRequires:  make
+BuildRequires:  gcc %{?with_gtest:gtest-devel}
+%if %{with lz4}
+BuildRequires:  lz4-devel
+%endif
+%if %{with lzma}
+BuildRequires:  xz-devel
+%endif
+%if %{with pzstd}
+BuildRequires:  gcc-c++
+%endif
+%if %{with zlib}
+BuildRequires:  zlib-devel
+%endif
+BuildRequires:  execstack
 
 %description
-The zstd package contains programs for compressing and decompressing files
+Zstd, short for Zstandard, is a fast lossless compression algorithm,
+targeting real-time compression scenarios at zlib-level compression ratio.
 
-%package        devel
-Summary:        Header and development files for zstd
-Requires:       %{name} = %{version}-%{release}
-Provides:       lib%{name}-devel = %{version}-%{release}
+%package -n lib%{name}
+Summary:        Zstd shared library
 
-%description    devel
-Header and development files for zstd compression
+# For backwards compatibility
+Provides:       zstd-libs = %{version}-%{release}
+Obsoletes:      zstd-libs <= 1.5.5-5
 
-%package        libs
-Summary:        Libraries for zstd
-Group:          System Environment/Libraries
-Provides:       lib%{name} = %{version}-%{release}
+%description -n lib%{name}
+Zstandard compression shared library.
 
-%description    libs
-This package contains minimal set of shared zstd libraries.
+%package -n lib%{name}-devel
+Summary:        Header files for Zstd library
+Requires:       lib%{name}%{?_isa} = %{version}-%{release}
 
-%package        doc
-Summary:        Documentation files for zstd
-Requires:       %{name} = %{version}-%{release}
+# For backwards compatibility
+Provides:       zstd-devel = %{version}-%{release}
+Obsoletes:      zstd-devel <= 1.5.5-5
 
-%description    doc
-Documentation files for zstd
+%package -n lib%{name}-static
+Summary:        Static variant of the Zstd library
+Requires:       lib%{name}-devel = %{version}-%{release}
+
+%description -n lib%{name}-devel
+Header files for Zstd library.
+
+%description -n lib%{name}-static
+Static variant of the Zstd library.
 
 %prep
-%autosetup
+%setup -q
```

Comment:
Why the revert? We try to stick to the new `%autosetup` wherever possible.

Reply:
reduce delta from inherited spec